lib/std/map.mth

```diff
@@ -28,3 +28,53 @@ struct Map(k,v) {
         lookup-pair(keq) map(value)
     }
 }
+
+
+struct +KeyValuePair(k, +v) {
+    key: k
+    +value: +v
+    --
+    def +for(f) [ (*a +v -- *b) *a +KeyValuePair(k, +v) -- *b ] {
+        /+KeyValuePair key> drop +value> f 
+    }
+
+    def rdrop(del {rdrop}) { +for(del) }
+}
+
+struct +Map(k, +v) {
+    +pairs: +List(+KeyValuePair(k, +v)) 
+    --
+    def init [ -- +Map(k,+v) ] {
+        +List.+Nil >+pairs +Map
+    }
+    def +insert [ +KeyValuePair(k,+v) +Map(k,+v) -- +Map(k,+v) ] { +pairs:+Cons }
+
+    def +delete (keq {==}, del {rdrop}) [ 
+        (*c k k -- *c Bool, *c +v -- *c) 
+        *c k +Map(k, +v) -- *c +Map(k, +v)
+    ] {
+        +pairs:+filter-for(dup dip_(key rdip:keq), dip:+for:del) drop
+    }
+
+    def +find-pair(keq {==}, f, g) [ 
```

Comment:
Default arguments should come after non-default ones. This is probably accepted right now by the compiler but it's a bug.

```suggestion
    def +find-pair(f, g, keq {==}) [ 
```

Reply:
Same with the other words below. (Also in the type signature, the parameter order needs to be updated as well.)